frontend/packages/console-app/src/actions/creators/deployment-factory.ts

```diff
@@ -0,0 +1,61 @@
+import i18next from 'i18next';
+import { Action } from '@console/dynamic-plugin-sdk';
+import { configureUpdateStrategyModal, errorModal } from '@console/internal/components/modals';
+import { togglePaused, asAccessReview } from '@console/internal/components/utils';
+import { K8sResourceKind, K8sKind } from '@console/internal/module/k8s';
+import { resourceLimitsModal } from '../../components/modals/resource-limits';
+import { ResourceActionFactory } from './common-factory';
+
+export const DeploymentActionFactory: ResourceActionFactory = {
+  EditDeployment: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: `edit-deployment`,
+    label: i18next.t('console-app~Edit {{kind}}', { kind: kind.kind }),
+    cta: {
+      href: `/edit-deployment/ns/${obj.metadata.namespace}?name=${obj.metadata.name}&kind=${kind.kind}`,
+    },
+    // TODO: Fallback to "View YAML"? We might want a similar fallback for annotations, labels, etc.
+    accessReview: asAccessReview(kind, obj, 'update'),
+  }),
+  UpdateStrategy: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: 'edit-update-strategy',
+    label: i18next.t('console-app~Edit update strategy'),
+    cta: () => configureUpdateStrategyModal({ obj }),
```

Comment:
Careful with these migrations when it is an object param. `configureUpdateStrategyModal` wants a prop named `deployment` and not `obj`

Reply:
Sorry, missed that one. Thanks for catching this.